js/ui/date_box/ui.date_box.base.js

```diff
@@ -62,6 +62,10 @@ const DateBox = DropDownEditor.inherit({
         return extend(this.callBase(), this._strategy.supportedKeys());
     },
 
+    _shouldCloseOnTab() {
+        return this._isInstantlyMode() && !this.option('type') === 'datetime';
```

Comment:
This function always returns `false`.
```suggestion
        return this._isInstantlyMode() && this.option('type') !== 'datetime';
```
Don't forget to add a test

Reply:
Test: https://github.com/DevExpress/DevExtreme/pull/24115/files#diff-8bf0c221d7c0f89e0369d7b346a4a41a4ac5184bd52039c4f166f7bdff02aa91R4780